Service event logging service

andrejn-ct opened the pull request:
A new logging service is implemented. It listens to all the service events and logs them into a database.
At the moment no cleanup is in place, meaning that the table containing the log entries just grows. Some way to manually delete the entries would be needed. Perhaps some sort of housecleaning with a dedicated time-to-live parameter for the logging service.

andrejn-ct committed:
Service events are enabled for most (almost all) services. Basic tests are also implemented.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Service events are enabled for most (almost all) services. Basic tests are also implemented.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Service events are enabled for most (almost all) services. Basic tests are also implemented.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Service events are enabled for most (almost all) services. Basic tests are also implemented.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Service events are enabled for most (almost all) services. Basic tests are also implemented.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Service events are enabled for most (almost all) services. Basic tests are also implemented.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Service events are enabled for most (almost all) services. Basic tests are also implemented.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Service events are enabled for most (almost all) services. Basic tests are also implemented.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

riccardomodanese committed:
service events - fix event context lifecycle

Signed-off-by: riccardomodanese <riccardo.modanese@eurotech.com>

andrejn-ct committed:
Fix for post merge problems.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Removed some obsolete log messages.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Reorganization of tests for service event hadlers.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Reorganization of tests for service event hadlers.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Added the processing of group events to the device service.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Added the processing of group events to the device service.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Added liquibase scripts to drop redundant foreign key constraints.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Added liquibase scripts to drop redundant foreign key constraints.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Replaced kapua locator static initializers in the service implementations.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Replaced kapua locator static initializers in the service implementations.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Replaced kapua locator static initializers in the service implementations.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Replaced kapua locator static initializers in the service implementations.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Replaced kapua locator static initializers in the service implementations.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Removed redundant dependency from console job module pom file

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Added service event handlers to the trigger service

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Added service event handlers to the trigger service

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Service event handling in the Trigger service. Added also a basic event handling test.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Service event handling in the Trigger service. Added also a basic event handling test.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Service event handling in the Trigger service. Added also a basic event handling test.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Added some test steps for the Trigger service event handling.-s

andrejn-ct committed:
Added some test steps for the Trigger service event handling.-s

andrejn-ct committed:
Updated the cleanup SQL script for integration tests

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Fix for failing tests after rebase

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Fix for failing tests after rebase

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Copy-paste error in some log strings.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Split the Job service event test into separate test suite.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Fixed some tst cases to work with the new job step numbering restrictions.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Fixed Job Scheduler service event handler.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Service event handler for the EndpointInfo service.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Service event handler for the EndpointInfo service.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Service event handler for the Endpoint Info service. Fix for service instantiation issues.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Fix for a small mistake in the Endpoint Info service event hander.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Endpoint info service integration test skeleton. Single service event test implemented.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Fixed a event handling issue in the EndpointInfo service. Added a specific test case.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Improved database cleanup between test scenarios.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Added a basic job service test.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Added a basic job service test.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Added some job scheduler related tests.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Added some job scheduler related tests.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Added some job scheduler related tests.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Some additional scheduler service tests.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Some additional scheduler service tests.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Remove service event handling from the domain registry service as it is unnecessary.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Improvement to logging. In case of service bus anomalies, the handling services log a warning.-s

andrejn-ct committed:
Fix for merge induced errors

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Fix for merge induced errors

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Service event logger service. Initial implementation.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Fix for rebase issues.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Changed the service event logging service to non resurce limited.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Extended the privileged context to the whole event logging method.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Cleaned up the configuration of the Job service unit test steps.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Some additional cleanup of the job service test setup.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Improved the handling of service events. All clean up actions are performed in a privileged context.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Replaced event source name literals with constants.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Replaced event source name literals with constants.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Replaced event source name literals with constants.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Moved the javadoc comments fo rthe event osurce names to the common constants base class.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Fixed the javadoc comments for a better documentation.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

Coduz commented on service/eventlog/api/src/main/java/org/eclipse/kapua/service/eventlog/EventLog.java:
```diff
@@ -0,0 +1,142 @@
+     *
+     * @return
+     */
+    @XmlElement(name = "entityScopeId")
```
This is missing `@XmlJavaTypeAdapter(KapuaIdAdapter.class)`

Coduz commented on service/eventlog/api/src/main/java/org/eclipse/kapua/service/eventlog/EventLog.java:
```diff
@@ -0,0 +1,142 @@
+     * @return
+     */
+    @XmlElement(name = "entityId")
+    KapuaId getEntityId();
```
@XmlJavaTypeAdapter(KapuaIdAdapter.class)

Coduz commented on service/eventlog/api/src/main/java/org/eclipse/kapua/service/eventlog/EventLogCreator.java:
```diff
@@ -0,0 +1,135 @@
+     * @return
+     */
+    @XmlElement(name = "entityScopeId")
+    KapuaId getEntityScopeId();
```
@XmlJavaTypeAdapter(KapuaIdAdapter.class)

Coduz commented on service/eventlog/api/src/main/java/org/eclipse/kapua/service/eventlog/EventLogCreator.java:
```diff
@@ -0,0 +1,135 @@
+     * @return
+     */
+    @XmlElement(name = "entityId")
+    KapuaId getEntityId();
```
@XmlJavaTypeAdapter(KapuaIdAdapter.class)

Coduz commented on service/eventlog/internal/src/main/java/org/eclipse/kapua/service/eventlog/internal/EventLogServiceImpl.java:
```diff
@@ -0,0 +1,194 @@
+        // Check existence
+        EventLog eventLog = find(scopeId, eventLogId);
+        if (eventLog == null) {
+            throw new KapuaEntityNotFoundException(eventLog.TYPE, eventLogId);
```
This will cause a `NullPointerException`. It should be `EventLog.TYPE`

Coduz commented on service/eventlog/internal/src/main/java/org/eclipse/kapua/service/eventlog/internal/EventLogServiceImpl.java:
```diff
@@ -0,0 +1,194 @@
+    @ListenServiceEvent(fromAddress = "job")
+    @ListenServiceEvent(fromAddress = "group")
+    public void onKapuaEvent(ServiceEvent kapuaEvent) throws KapuaException {
+        if (kapuaEvent == null) {
```
`throw` some exception otherwise `logReceivedEvent` will throw a `NullPointerException`

Coduz commented on service/eventlog/internal/src/main/java/org/eclipse/kapua/service/eventlog/internal/EventLogServiceImpl.java:
```diff
@@ -0,0 +1,194 @@
+    @Override
+    public EventLog create(EventLogCreator eventLogCreator) throws KapuaException {
+        //
+        // Argument Validation
```
On argument validation please remove the `.get()` part form Id checks. 
We check on `.getScopeId` which, when `null`, is an invalid parameter.

Coduz reviewed (changes requested):
Hi Andrej, 

I've made a quick code review. 
There are some things to change. 
Then I'll also try to test, if required.

Regards, 

_Alberto_

andrejn-ct committed:
Minor fixes to address Albertos comments.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Rebase to the latest state of develop.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Rebase on top of current develop head.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Fix for rebase issues

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Missing cucumber steps will be now considered test failures.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Event log configurations.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Event logging can now be disabled.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Fixes for rebase issues.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Added some missing event broker startup steps.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Added a purging method for obsolete event log entries.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Fix for merge conflicts with ancestor develop head.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Refactored the event logger service folder structure.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Added the skeleton for the event log manager module.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Laid out the skeleton for the EventLog housekeeper module. Purge works. Scheduling still needs to be implemented.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
The log purging logic was moved to the EventLog HouseKeeper module.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Fixed some device service tests and test steps.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
EventLogDAO is now aligned with the new DAO interface

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Removed an unnecessary JAXB context provider definition from the event log service

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
EventLog housekeeping. Added entry pagination in the purge method.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

Commit:
Manage sortFields unmarshalling in REST Datastore queries

Signed-off-by: Claudio Mezzasalma <claudio.mezzasalma@eurotech.com>

andrejn-ct committed:
Updated cucumber features to use new embedded component startup.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
The number of event log entries to be deleted in a single batch is now configurable.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

Commit:
Move JsonMessageQuery to rest-api-resources module

Signed-off-by: Claudio Mezzasalma <claudio.mezzasalma@eurotech.com>

andrejn-ct committed:
Aligned common service event plumbing back with upstream develop.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

Commit:
Write correct HTTP error code in REST Internal Error

Signed-off-by: Claudio Mezzasalma <claudio.mezzasalma@eurotech.com>

Commit:
Fix Exception Message with no Resource Bundle

Signed-off-by: Claudio Mezzasalma <claudio.mezzasalma@eurotech.com>

muros-ct committed:
Changed port of embedded Jetty server (#2161)

Because it interfered with jenkins.

Signed-off-by: Uros Mesaric Kunst <uros.mesaric-kunst@comtrade.com>

Coduz committed:
Merge pull request #2158 from lorthirk/fix-httpStatusInternalError

Write correct HTTP error code in REST Internal Error

Coduz committed:
Merge pull request #2159 from lorthirk/fix-exception_no_message_bundle

Fix Exception Message with no Resource Bundle

muros-ct committed:
Fixing Jenkins Rest API tests of Kapua (#2168)

Added stacktrace at H2 DB helper tcp connector start.

Added setting for H2 to bind to localhost.

Signed-off-by: Uros Mesaric Kunst <uros.mesaric-kunst@comtrade.com>

ct-ajovanovic committed:
Fix for Submit button not enabled on ComboBox value change (#2162)

Signed-off-by: ct-ajovanovic <aleksandra.jovanovic@comtrade.com>

ct-ajovanovic committed:
Changed the field type for the Command Output (#2169)

Signed-off-by: ct-ajovanovic <aleksandra.jovanovic@comtrade.com>

Coduz committed:
Fixed scopeId used to fetch all roles available

Signed-off-by: coduz <alberto.codutti@eurotech.com>

Coduz committed:
Fixed visualization of AccessRole in description tab

Signed-off-by: coduz <alberto.codutti@eurotech.com>

Coduz committed:
Merge pull request #2171 from eclipse/fix-userRoleAddForm

Fix user role add form

Commit:
Stop Endpoint Info fallback if there are Endpoints specified in scope

Signed-off-by: Claudio Mezzasalma <claudio.mezzasalma@eurotech.com>

andrejn-ct committed:
Fixed event broker startup in various cucumber feature files.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

andrejn-ct committed:
Tweaked the delays in the tenant service event tests

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

Commit:
Added check also in count
Also removed duplicated code

Signed-off-by: Claudio Mezzasalma <claudio.mezzasalma@eurotech.com>

andrejn-ct committed:
Added test groups to the travis build.

Signed-off-by: Andrej Nussdorfer <andrej.nussdorfer@comtrade.com>

Commit:
Managing offline Device errors fix

Signed-off-by: ct-ajovanovic <aleksandra.jovanovic@comtrade.com>

Commit:
Remove slf4j and logback jar exclusion from pom.xml

Signed-off-by: Claudio Mezzasalma <claudio.mezzasalma@eurotech.com>

Commit:
Add @ApiModelProperty in CredentialCreator.credentialKey

Signed-off-by: Claudio Mezzasalma <claudio.mezzasalma@eurotech.com>

Coduz committed:
Merge pull request #2152 from ct-ajovanovic/fix-issue2117

Managing offline Device errors fix

Coduz committed:
Merge pull request #2182 from lorthirk/fix-swagger_ui_credentials_plainKey

Fix swagger ui credentials plain key

Coduz committed:
Merge pull request #2173 from lorthirk/change-stop_endpoint_fallback_condition

Stop Endpoint Info fallback if there are Endpoints specified in scope

Coduz committed:
Merge pull request #2121 from lorthirk/feature-sortfields_rest_datastore_query

Manage sortFields unmarshalling in REST Datastore queries

ct-ajovanovic committed:
Child account name not in error state fix (#2179)

Signed-off-by: ct-ajovanovic <aleksandra.jovanovic@comtrade.com>

Coduz committed:
Added JobStepDefinition filtering on add dialog

Signed-off-by: coduz <alberto.codutti@eurotech.com>

Coduz committed:
Merge pull request #2191 from eclipse/fix-jobStepDefinitionFiltering

Added JobStepDefinition filtering on add dialog

ct-ajovanovic committed:
Role paging in Users fix (#2189)

Signed-off-by: ct-ajovanovic <aleksandra.jovanovic@comtrade.com>

andrejn-ct commented:
Just a test pull request. Not to be merged.
